Sources/SwiftParser/Specifiers.swift

```diff
@@ -415,7 +441,114 @@ extension RawAccessorEffectSpecifiersSyntax: RawEffectSpecifiersTrait {
       }
     }
   }
+}
+
+extension RawDeinitEffectSpecifiersSyntax: RawMisplacedEffectSpecifiersTrait {
+  enum MisspelledAsyncTokenKinds: TokenSpecSet {
+    case await
+    case reasync
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.await, allowAtStartOfLine: false): self = .await
+      case TokenSpec(.reasync): self = .reasync
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .await: return TokenSpec(.await, allowAtStartOfLine: false)
+      case .reasync: return .keyword(.reasync)
+      }
+    }
+  }
+
+  enum CorrectAsyncTokenKinds: TokenSpecSet {
+    case async
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.async): self = .async
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .async: return .keyword(.async)
+      }
+    }
+  }
+
+  enum MisspelledThrowsTokenKinds: TokenSpecSet {
+    case `rethrows`
+    case `try`
+    case `throw`
+    case `throws`
 
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.rethrows): self = .rethrows
+      case TokenSpec(.try, allowAtStartOfLine: false): self = .try
+      case TokenSpec(.throw, allowAtStartOfLine: false): self = .throw
+      case TokenSpec(.throws): self = .throws
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .rethrows: return .keyword(.rethrows)
+      case .try: return TokenSpec(.try, allowAtStartOfLine: false)
+      case .throw: return TokenSpec(.throw, allowAtStartOfLine: false)
+      case .throws: return .keyword(.throws)
+      }
+    }
+  }
+
+  enum CorrectThrowsTokenKinds: TokenSpecSet {
+    // Uninhabited
+
+    init?(lexeme: Lexer.Lexeme) {
+      return nil
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      }
+    }
+  }
+
+  var throwsSpecifier: RawTokenSyntax? { nil }
+
+  init(
+    asyncSpecifier: RawTokenSyntax?,
+    throwsSpecifier: RawTokenSyntax?,
+    arena: __shared SwiftSyntax.SyntaxArena
+  ) {
+    // Inserted missing throws is discarded
```

Comment:
Just as I’m reading this. I would add the following comment
```
`throwsSpecifier` should never be present because `CorrectThrowsTokenKinds` is an empty `TokenSpecSet`.
```

Reply:
Not even because of that, but because `parseMisplacedEffectSpecifiers()` only creates missing tokens

Reply:
Ah, I see. Thanks for clearing up the misunderstanding. I think it would be good to mention that as a comment.